src/query/graphite/storage/m3_wrapper.go

```diff
@@ -141,19 +141,28 @@ func truncateBoundsToResolution(
 
 func translateTimeseries(
 	ctx xctx.Context,
-	result *storage.FetchResult,
+	result block.Result,
 	start, end time.Time,
 ) ([]*ts.Series, error) {
-	m3list := result.SeriesList
-	series := make([]*ts.Series, len(m3list))
+	block := result.Blocks[0]
```

Comment:
Should there be a bounds check here to make sure there's a block? (i.e. at least len > 0)

Reply:
Good catch